src/search/plugins/plugin.h

```diff
@@ -111,7 +112,12 @@ class TypedFeature : public FeatureAuto<Constructed> {
     }
 
     Any construct(const Options &options, const utils::Context &context) const override {
-        std::shared_ptr<Base> ptr = this->create_component(options, context);
+        std::shared_ptr<Base> ptr;
+        try {
+            ptr = this->create_component(options, context);
```

Comment:
(Why) do we need `this` here?

Reply:
It is needed 
https://isocpp.org/wiki/faq/templates#nondependent-name-lookup-members

Reply:
I like that the description starts with "This might hurt your head; better if you sit down."